src/shared/serverinfo.hpp

```diff
@@ -0,0 +1,231 @@
+
+    int m_lastping;
+    int m_nextping;
+    int m_pings[MAXPINGS];
+    int m_lastinfo;
+
+    // UDP? port number, probably redundant with m_address
+    int m_port;
+
+    int m_priority;
+    int m_ping;
+
+    // name resolution state.
+    enum
+    {
+        UNRESOLVED,
+        RESOLVING,
+        RESOLVED
+    } m_resolved;
+
+    // is "servers" sorted?
+    static bool s_sortedservers;
+
+    const char* description( void ) const;
+    void clearpings();
+    void cleanup();
+    void addping(int rtt, int millis);
+    int protocol_version( void ) const;
+    SSTAT server_status( void ) const;
+public:
+    serverinfo(uint ip, int port, int priority = 0);
+    ~serverinfo();
+
+    // marks server as not queried and flags servers(list) as not sorted
+    void reset();
+
+    // compare 2 servers according to a sorting "style".
+    // returns 0 if style is invalid, abort() in debug mode.
+    //
+    // Styles NUMPLRS and PRIO are "better if higher".
+    // Others are "better if lower".
+    // Reverse invert the sorting.
+    //
+    // returns:
+    //   * -1 if *this < other
+    //   *  0 if *this = other
+    //   *  1 if *this > other
+    int compare( serverinfo const& other, int style, bool reverse ) const;
+    int compare( serverinfo const& other, SINFO style, bool reverse ) const;
+
+    // return 0 if versions match, -1 if *this > other and 1 otherwise
+    int version_compare( serverinfo const& other ) const;
+
+    // returns server name.
+    // TODO: remove it. Used by:
+    //   _ resolverquery() in engine/serverbrowser.cpp
+    //   _ servercompare() in game/client.cpp
+    char const* name( void ) const;
+
+    // "return" a property of this server to CS
+    // if property < 0: returns the number of properties one can query (4)
+    // if property > 3: do nothing.
+    // if property > 0:
+    //   if index < 0: returns the size of requested array
+    //   if index > array.size(): do nothing
+    //   if property == 1: returns m_attr[index] (integer)
+    //   if property == 2: returns players[index] (result?)
+    //   if property == 3: returns handles[index] (result?)
+    // if propert == 1:
+    //   if index ==  0: returns server_status() (integer)
+    //   if index ==  1: returns name (result) (this is probably the IP or domain name)
+    //   if index ==  2: returns port (integer)
+    //   if index ==  3: returns description or name if no description (result)
+    //   if index ==  4: returns map played (result)
+    //   if index ==  5: returns number of players (integer)
+    //   if index ==  6: returns ping? Guess it's roundtrip time? (integer)
+    //   if index ==  7: returns m_lastinfo? guess it's time of last update? (integer)
+    //   if index ==  8: returns authhandle?? (result)
+    //   if index ==  9: returns flags??  (result)
+    //   if index == 10: returns branch?? (resull)
+    //   if index == 11: returns priority (integer)
+    void cube_get_property( int property, int index );
```

Comment:
What does `cube` refer to? The docstring implies this method is somewhat useful with regards to CubeScript. `cube` != `CubeScript`, `cube` is a game whereas CubeScript is a language. I find it a bit confusing.

Other than that, there should be a TODO to invite further contributors into reverse engineering the other values and answer the open questions.